src/server/frontend/wayland/wayland_connector.cpp

```diff
@@ -2398,6 +2442,7 @@ mf::WaylandConnector::WaylandConnector(
         display.get(),
         display_config);
     shell_global = std::make_unique<mf::WlShell>(display.get(), shell, *seat_global);
+    data_device_manager_global = std::make_unique<DataDeviceManager>(display.get(), wl_data_device_interface.version);
```

Comment:
I don't think this is safe - if the wl_data_device protocol ever gets a new version, the version we claim to support will change, without us actually supporting the new version.

At best the new version has a new method, and clients which call that method crash the server. At worst, there's a purely semantic change which results in some clients mysteriously failing to work properly. 😁

We need to always explicitly write our max_version; any change to that number must require code changes anyway.